### PR TITLE
DRY out query translation failure in tests

### PR DESCRIPTION
**Pull request by roji**

### CODE REVIEW

**Comment by maumar** [comment]
We have a method `AssertTranslationFailed` - we should DRY/combine those. Perhaps move to query test base?

**Comment by roji** [comment]
New version up.

File: test/EFCore.Relational.Specification.Tests/Query/QueryNoClientEvalTestBase.cs [hunk 1]
```diff
@@ -205,7 +205,7 @@ private string NormalizeDelimitersInRawString(string sql)
 
         private void AssertTranslationFailed(Action testCode)
         {
-            Assert.Contains(
+            Assert.StartsWith(
```
Comment (maumar) [inline test/EFCore.Relational.Specification.Tests/Query/QueryNoClientEvalTestBase.cs]: revert to Contains

File: test/EFCore.Specification.Tests/Query/QueryTestBase.cs [hunk 2]
```diff
@@ -1149,6 +1150,12 @@ protected static IEnumerable<TResult> MaybeDefaultIfEmpty<TResult>(IEnumerable<T
                 : caller.DefaultIfEmpty();
         }
 
+        protected static async Task AssertTranslationFailed(Func<Task> query)
+            => Assert.StartsWith(
```
Comment (smitpatel) [inline test/EFCore.Specification.Tests/Query/QueryTestBase.cs]: Contains

**Pull request merged**